tiledb/sm/misc/types.h

```diff
@@ -38,6 +38,7 @@
 #include <string>
 #include <vector>
 #include "tiledb/common/logger_public.h"
+#include "tiledb/sm/cpp_api/exception.h"
```

Comment:
We can't use external headers inside the library. If you need the type check please re-implement it.

Reply:
The include is for the TileDBError exception.

Reply:
The `Status*` classes are going to be converted to support being thrown soon. For now use one of the stl exception types to avoid this include.

Reply:
Yes, yes, I see now what I did wrong, first architecture practical lesson, sorry. I updated the PR.